server-spi-private/src/main/java/org/keycloak/services/clientpolicy/ClientPoliciesUtil.java

```diff
@@ -0,0 +1,747 @@
+            profilesJson = session.clientPolicy().getClientProfilesOnKeycloakApp();
+        }
+
+        // deserialize existing profiles (json -> representation)
+        if (profilesJson == null) {
+            return new ClientProfilesRepresentation();
+        }
+        try {
+            profilesRep = convertClientProfilesJsonToRepresentation(profilesJson);
+        } catch (ClientPolicyException e) {
+            throw new ClientPolicyException(e.getError(), e.getErrorDetail());
+        }
+        if (profilesRep == null) {
+            return new ClientProfilesRepresentation();
+        }
+
+        return profilesRep;
+    }
+
+    /**
+     * gets existing client profiles in a realm as model.
+     * not return null.
+     */
+    public static Map<String, ClientProfileModel> getClientProfilesModel(KeycloakSession session, RealmModel realm) {
+        // get existing profiles as json
+        String profilesJson = session.clientPolicy().getClientProfilesJsonString(realm);
+        if (profilesJson == null) {
+            return Collections.emptyMap();
+        }
+
+        // deserialize existing profiles (json -> representation)
+        ClientProfilesRepresentation profilesRep = null;
+        try {
+            profilesRep = convertClientProfilesJsonToRepresentation(profilesJson);
+        } catch (ClientPolicyException e) {
+            logger.warnv("{0} Failed to serialize client profiles json string. err={1}, errDetail={2}", LOGMSG_PREFIX, e.getError(), e.getErrorDetail());
+            return Collections.emptyMap();
+        }
+        if (profilesRep == null || profilesRep.getProfiles() == null) {
+            return Collections.emptyMap();
+        }
+
+        // constructing existing profiles (representation -> model)
+        Map<String, ClientProfileModel> profileMap = new HashMap<>();
+        for (ClientProfileRepresentation profileRep : profilesRep.getProfiles()) {
+            // ignore profile without name
+            if (profileRep.getName() == null) {
+                continue;
+            }
+
+            ClientProfileModel profileModel = new ClientProfileModel();
+            profileModel.setName(profileRep.getName());
+            profileModel.setDescription(profileRep.getDescription());
+            if (profileRep.isBuiltin() != null) {
+                profileModel.setBuiltin(profileRep.isBuiltin().booleanValue());
+            } else {
+                profileModel.setBuiltin(false);
+            }
+
+            if (profileRep.getExecutors() == null) {
+                profileModel.setExecutors(new ArrayList<>());
+                profileMap.put(profileRep.getName(), profileModel);
+                continue;
+            }
+
+            List<Object> executors = new ArrayList<>();
+            if (profileRep.getExecutors() != null) {
+                profileRep.getExecutors().stream().forEach(obj->{
+                    ObjectMapper mp = new ObjectMapper();
```

Comment:
Is it possible to re-use some existing ObjectMapper? Ideally for example `JsonSerialization.mapper`? The instantiation of ObjectMapper has some performance costs as it is instantiating lots of other fasterxml common objects/factories under the covers.

I see you used `new ObjectMapper()` in many places in this PR, so I suggest to fix all of them

Reply:
@mposolda Yes, I will use `static final ObjectMapper` of this `ClientPoliciesUtil`. Also I will find out places using multiple `new ObjectMapper()` and replace it with `static final ObjectMapper`.

Reply:
Thanks